app/src/main/java/com/example/bar/foo/myapplication/MainActivity.java

```diff
@@ -17,22 +21,43 @@
         System.loadLibrary("native-lib");
     }
 
+
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
 
-        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
-        fab.setOnClickListener(new View.OnClickListener() {
+        final Timer timer = new Timer();
+        final TimerTask timerTask = new TimerTask() {
+            @Override
+            public void run() {
+                farmPantsu();
+                //updateStatus(); how the heck do threads work again
+            }
+        };
+
+        FloatingActionButton fabManual = (FloatingActionButton) findViewById(R.id.fabManual);
+        fabManual.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                fetchPantsu();
+                fetchPantsu(); // Maybe show some pantsu images in the background as you click, with a golden one when it's an ssr
                 updateStatus();
             }
         });
 
+        FloatingActionButton fabBuyFarmer = (FloatingActionButton) findViewById(R.id.fabBuyFarmer);
+        fabBuyFarmer.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View view){
+                buyFarmer();
+                if(getFarmers()==1)
```

Comment:
I think this will cause the timer to start multiple times if you have 1 farmer, but not enough to buy a second.